crates/red_knot_module_resolver/src/path.rs

```diff
@@ -725,6 +725,82 @@ impl PartialEq<ModuleResolutionPathRef<'_>> for VendoredPathBuf {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub(crate) struct ModuleSearchPath(Arc<ModuleResolutionPathBuf>);
+
+impl ModuleSearchPath {
+    pub(crate) fn extra(path: SystemPathBuf) -> Option<Self> {
+        Some(Self(Arc::new(ModuleResolutionPathBuf::extra(path)?)))
+    }
+
+    pub(crate) fn first_party(path: SystemPathBuf) -> Option<Self> {
+        Some(Self(Arc::new(ModuleResolutionPathBuf::first_party(path)?)))
+    }
+
+    pub(crate) fn custom_stdlib(path: &SystemPath) -> Option<Self> {
+        Some(Self(Arc::new(ModuleResolutionPathBuf::standard_library(
+            FilePath::System(path.join("stdlib")),
+        )?)))
+    }
+
+    pub(crate) fn vendored_stdlib() -> Self {
+        Self(Arc::new(ModuleResolutionPathBuf(
+            ModuleResolutionPathBufInner::StandardLibrary(FilePath::vendored("stdlib")),
+        )))
+    }
+
+    pub(crate) fn site_packages(path: SystemPathBuf) -> Option<Self> {
+        Some(Self(Arc::new(ModuleResolutionPathBuf::site_packages(
+            path,
+        )?)))
+    }
+
+    pub(crate) fn editable(system: &dyn System, path: SystemPathBuf) -> Option<Self> {
+        Some(Self(Arc::new(
+            ModuleResolutionPathBuf::editable_installation_root(system, path)?,
+        )))
+    }
```

Comment:
I think we should inline the 

https://github.com/astral-sh/ruff/blob/dc8ae368ff6a5dedd7e623393f700c8709362cb6/crates/red_knot_module_resolver/src/path.rs#L171-L212

and remove the definitions from `ModuleResolutionPathBuf`. The methods on `ModuleResolutionPathBuf` are only used in tests and I suspect that the tests should probably use `ModuleResolutionPath` instead.

Reply:
This makes sense but I'm going to postpone it for now. The idea of this PR was that it would be a small, incremental change that would be easy to review. Addressing this would require lots of changes to tests, and I think this PR is an improvement over the status quo as it stands.